File: 1927-sum-game/1927-sum-game.py
class Solution:
    def sumGame(self, A: str) -> bool:
        s, q = [0, 0], [0, 0]
        n = len(A)
        for i in range(n):
            j = i // (n // 2)
            if A[i] == "?":
                q[j] += 1
            else:
                s[j] += int(A[i])
        return (q[0] + q[1]) & 1 == 1 or (s[0] - s[1]) != (q[1] - q[0]) * 4.5

File: 1927-sum-game/1927-sum-game.cpp
class Solution {
public:
    bool sumGame(string A) {
        int sum[2] = {0, 0}, q[2] = {0, 0};
        int n = A.length();
        for (int i = 0; i < n; i++) {
            int j = i / (n >> 1);
            if (A[i] == '?')
                q[j]++;
            else
                sum[j] += A[i] - '0';
        }
        return ((q[0] + q[1]) & 1) ||
               ((sum[0] - sum[1]) << 1) != (q[1] - q[0]) * 9;
    }
};